Resources/Private/JavaScript/components/RedirectForm.tsx

```diff
@@ -0,0 +1,320 @@
+                            statusCode: defaultStatusCode,
+                            ...redirect,
+                            isSendingData: false,
+                        });
+                    }
+
+                    if (changedRedirects.length > 1) {
+                        const changeList = this.renderChangedRedirects(changedRedirects);
+                        notificationHelper.warning(message, changeList);
+                    } else {
+                        notificationHelper.ok(message);
+                    }
+                } else {
+                    notificationHelper.error(message);
+                    this.setState({
+                        isSendingData: false,
+                    });
+                }
+            }).catch(error => {
+                notificationHelper.error(error);
+                this.setState({
+                    isSendingData: false,
+                });
+            }
+        );
+    };
+
+    /**
+     * Stores any change to the form in the state
+     *
+     * @param event
+     */
+    handleInputChange = (event: ChangeEvent): void => {
+        const target: HTMLInputElement = event.target as HTMLInputElement;
+        const {name, value} = target;
+        this.setState({
+            [name]: value.substring(0, MAX_INPUT_LENGTH),
+        });
+    };
+
+    /**
+     * Stores changes to datetime fields in the state
+     *
+     * @param property
+     * @param datetime
+     */
+    private handleDatePickerChange(property: string, datetime: Date | string) {
+        const formattedValue = typeof datetime === 'string' ? datetime : formatReadable(datetime);
+        this.setState({
+            [property]: formattedValue,
+        });
+    }
+
+    /**
+     * Renders a datepicker
+     *
+     * @param property
+     * @param dateTimeString
+     * @param placeholder
+     */
+    private renderDatePicker = (property: string, dateTimeString: string, placeholder: string) => {
+        const {translate} = this.props;
+        const dateTime = dateTimeString ? new Date(dateTimeString) : null;
+
+        return (
+            <DatePicker
+                dateFormat="yyyy-MM-dd HH:mm"
+                timeFormat="HH:mm"
+                showTimeSelect
+                timeIntervals={15}
+                todayButton={translate('datepicker.today', 'Today')}
+                placeholderText={placeholder}
+                selected={dateTime}
+                timeCaption={translate('datepicker.time', 'Time')}
+                onChange={(value) => this.handleDatePickerChange(property, value)}/>
+        );
+    };
+
+    /**
+     * Renders list of changed redirects to be used in a flash message
+     *
+     * @param changedRedirects
+     */
+    private renderChangedRedirects = (changedRedirects: Array<Redirect>): string => {
+        const {translate} = this.props;
+        return (
+            '<p>' + translate('message.relatedChanges', 'Related changes') + '</p>' +
```

Comment:
Never saw writing jsx as string representation.
Why not like that?
```
return (
  <> // Also called React.Fragment
    <p>{translate('message.relatedChanges', 'Related changes')}</p>
    <ul></ul>
  </>
);
```

Reply:
I have to give a string to the FlashMessage api, therefore I cannot just give a ReactElement